mixer/adapter/stackdriver/trace/trace.go

```diff
@@ -182,26 +184,47 @@ func buildSpanData(val *tracespan.Instance, parentContext trace.SpanContext, spa
 	}
 
 	spanKind := trace.SpanKindServer
+	parentSpanID := parentContext.SpanID
+	spanID := spanContext.SpanID
 	if val.ClientSpan {
 		spanKind = trace.SpanKindClient
+		// If this is a client span and rewrite_id is true, deterministically create a new span ID and
+		// rewrite span id to that one. This id should also be used as server span's parent span id.
```

Comment:
do we need to check to see if they are a shared span or not? iirc, the `zipkin-gcp` code had a check to see if the span was marked as shared.

Reply:
Instead of implicitly rewriting id based on the shared span or not, the newly added field `rewrite_id` provides configurability to users.